Updated oclif and project structure

andre-vidal opened the pull request:
<!---
☝️ PR title should follow conventional commits (https://conventionalcommits.org)
-->

### 🔗 Linked issue
N/A

### ❓ CLI Refactoring
<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [x] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

<!-- Describe your changes in detail -->
The Frontier project has been refactored to support the following:
- updated oclif framework version
  - support space ` ` command separators instead of colons `:` 
    - e.g. `frontier vue add component` instead of `frontier vue:add:component`
  - command event hooks to add custom functionality at different stages in a command execution
  - other features can be viewed [here](https://github.com/oclif/oclif#-migrating-from-v1)
- parsing of a `.frontierrc` file that exists in the root of a frontier related project that describes the scope of the cli to leverage

- consolidating the CLIs for different domains into one master CLI 
  - these will be managed by folder structure instead of as separate packages
<!-- Why is this change required? What problem does it solve? -->

These changes will result in the following experience for users:
- shorter command structure
- partial commands will still work in an existing frontier project
- plugins for the separate domains will no longer need to be installed since all features of frontier will be available by default

<!-- If it resolves an open issue, please link to the issue here. For example "Resolves #1337" -->

### 📝 Checklist

<!-- Put an `x` in all the boxes that apply. -->
<!-- If your change requires a documentation PR, please link it appropriately -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

andre-vidal committed:
replace plugin vue with plugin-vue-v2

andre-vidal committed:
update

andre-vidal committed:
update vue plugin

andre-vidal committed:
update

andre-vidal committed:
update plugin-vue config

andre-vidal committed:
refactor frontier cli

andre-vidal committed:
commit

andre-vidal committed:
update tsconfig

andre-vidal committed:
1.0.1-beta.0

andre-vidal committed:
revert version

andre-vidal committed:
1.0.1-beta.0

andre-vidal committed:
update package name

andre-vidal committed:
update package version

andre-vidal committed:
0.0.1-beta.0

andre-vidal committed:
update commands, hooks and help

andre-vidal committed:
update readme and help output

andre-vidal committed:
update

andre-vidal committed:
update

andre-vidal committed:
0.0.1-beta.1

andre-vidal committed:
Update command structure and readme documentation

andre-vidal committed:
update help config

andre-vidal committed:
update linting

andre-vidal committed:
0.0.1-beta.2

andre-vidal committed:
update command help menu

andre-vidal committed:
update tsconfig info

andre-vidal committed:
0.0.1-beta.3

andre-vidal committed:
update vue contants

andre-vidal committed:
update mobile implementation to match linting configs

andre-vidal committed:
update build  config

andre-vidal committed:
0.0.1-beta.4

andre-vidal committed:
update mobile readme

andre-vidal committed:
update documentation

andre-vidal committed:
update adam documentatiion

andre-vidal committed:
update tests

andre-vidal committed:
update build

andre-vidal committed:
0.0.1-beta.5

andre-vidal committed:
update readme

andre-vidal committed:
update template version

andre-vidal committed:
update build

andre-vidal committed:
0.0.1-beta.6

andre-vidal committed:
update readme and help menu

andre-vidal committed:
update build

andre-vidal committed:
0.0.1-beta.7

forevergareth commented:
I see the addition of the new cli structure but why does that come at the cost of removing everything else in the project? Like the libraries, folder structure, example projects, package manager?

andre-vidal committed:
update ambiguous command handling

andre-vidal committed:
update build

andre-vidal committed:
0.0.1-beta.8

andre-vidal committed:
Ambiguois command handling (#39)

* update

* update init hook

* build update

* 0.0.1-beta.9

* Fixed bug where arguments were not being passed along to sub command handlers

---------

Co-authored-by: Davane Davis <ddavis@realdecoy.com>

andre-vidal committed:
update build

andre-vidal committed:
update readme

andre-vidal committed:
0.0.2-beta.1

andre-vidal committed:
Add Dotnet Namespace (#40)

* create dotnet namespace
- add create-project command

* dotnet: update dotnet template tag

* dotnet: update readme

andre-vidal committed:
rebuild

andre-vidal committed:
0.0.2-beta.2

andre-vidal committed:
update dotnet namespace to include migration commands

andre-vidal committed:
update next steps output and install command

andre-vidal committed:
update

andre-vidal committed:
0.0.2-beta.3

andre-vidal committed:
Add commands for dotnet scaffolding

andre-vidal committed:
rebuild

andre-vidal committed:
0.0.2-beta.4

andre-vidal committed:
update documentation

andre-vidal committed:
update dotnet template tag

andre-vidal committed:
Merge branch 'development' of github.com:realdecoy/frontier into development

andre-vidal committed:
0.0.2-beta.5

andre-vidal committed:
add gitignore

andre-vidal committed:
ignore folder

andre-vidal committed:
update

andre-vidal committed:
update

andre-vidal committed:
update dotnet tag

andre-vidal committed:
Merge branch 'development' of github.com:realdecoy/frontier into development

andre-vidal committed:
0.0.2-beta.6

andre-vidal committed:
update version

andre-vidal committed:
update build

andre-vidal committed:
0.0.2-beta.7

andre-vidal committed:
update template version

andre-vidal committed:
0.0.2-beta.8

andre-vidal committed:
Update help menu configuration for commands

andre-vidal committed:
update copy files method

andre-vidal committed:
update

andre-vidal committed:
0.0.2-beta.9

andre-vidal committed:
remove console log

andre-vidal committed:
update build

andre-vidal committed:
0.0.2-beta.10

andre-vidal committed:
Sepa 675: create pipeline for deploying docs to aws s3 bucket (#46)

* Update 1.testing.md

* Update 1.testing.md (#41)

* Create deploy.yml

* Sepa 675 create pipeline for deploying to aws s3 bucket (#42)

* Update 1.testing.md

* Create deploy.yml

* Update deploy.yml

* Sepa 675 create pipeline for deploying to aws s3 bucket (#43)

* Update 1.testing.md

* Create deploy.yml

* Update deploy.yml

* Update deploy.yml

* Update deploy.yml

* Update 1.testing.md

---------

Co-authored-by: rd-rwalker <128421455+rd-rwalker@users.noreply.github.com>

andre-vidal committed:
Merge branch 'development' of github.com:realdecoy/frontier into development

andre-vidal committed:
update workflows

andre-vidal committed:
update workflow

andre-vidal committed:
update workflow

andre-vidal committed:
update workflow

andre-vidal committed:
update docs

andre-vidal committed:
update docs

andre-vidal committed:
update docs

andre-vidal committed:
update docs

andre-vidal committed:
update docs

andre-vidal committed:
update docs

andre-vidal committed:
update docs

andre-vidal committed:
update pipeline and docs

andre-vidal committed:
update deployment action

andre-vidal committed:
update testing

rd-rwalker committed:
Update 1.testing.md (#48)

rd-rwalker committed:
Sepa 712 completed (#52)

* update frontier documentation

* update frontier deocumentation

---------

Co-authored-by: travelandcode <ryonwalkerjnr2001@gmail.com>

rd-rwalker committed:
SEPA-707 ticket completed (#49)

* test to see if frontier documentation is updated

* Update deploy.yml

* Update package.json

---------

Co-authored-by: travelandcode <ryonwalkerjnr2001@gmail.com>

rd-rwalker committed:
update frontier documentation (#50)

Co-authored-by: travelandcode <ryonwalkerjnr2001@gmail.com>
Co-authored-by: Andre Vidal <avidal@realdecoy.com>

rd-rwalker committed:
SEPA 711 completed (#51)

* update frontier documentation vue create command

* update frontier documenation vue

* test

* update

* Update 3.next-steps.md

* update frontier documentation for vue next-steps

* Update deploy.yml

---------

Co-authored-by: travelandcode <ryonwalkerjnr2001@gmail.com>
Co-authored-by: Andre Vidal <avidal@realdecoy.com>

rd-rwalker committed:
SEPA 708 was completed (#53)

* first commit

* test

* test

* test

* test

* test

* final test

* update urls in frontier documentation

---------

Co-authored-by: travelandcode <ryonwalkerjnr2001@gmail.com>

andre-vidal committed:
update build

rd-rwalker committed:
documentation links added to the dotnet cli (#55)

Co-authored-by: travelandcode <ryonwalkerjnr2001@gmail.com>

rd-rwalker committed:
remaining show help descriptions added (#57)

Co-authored-by: travelandcode <ryonwalkerjnr2001@gmail.com>

Davane committed:
fix: app.json and bitrise.yml replace in files with regex (#56)

Co-authored-by: Davane Davis <ddavis@realdecoy.com>

tbennett109 committed:
Sepa 674 establish ssl out of the box for dotnet project (#58)

* dev certs comman created

* Generate and trust SSL certificate for HTTPS command created

* code cleaning

* silent true for trust command for mac/linux

tbennett109 committed:
system requirements page added (#60)

Co-authored-by: tbennett109 <tbennett@ealdecoy.com>

tbennett109 committed:
requirements title to bold, .net removed from recommendation (#62)

Co-authored-by: tbennett109 <tbennett@ealdecoy.com>

tbennett109 committed:
about documentation updated (#61)

Co-authored-by: tbennett109 <tbennett@ealdecoy.com>

ttaylor92 committed:
Updated pipelines file for deployment to npm (#59)

* Updated pipelines file for deployment to npm

* update deploy actions

* update branches

---------

Co-authored-by: andre-vidal <avidal@realdecoy.com>

andre-vidal committed:
Release 0.0.2-beta.12

Raffique-RD committed:
Sepa 739 documentation for migrations (#64)

* updates migration page

* updates to page

* chnaged index number

---------

Co-authored-by: Raffique <raffiquemuir@yahoo.com>

andre-vidal committed:
Update deploy-package.yml

andre-vidal committed:
Update deploy-documentation.yml

Raffique-RD committed:
update of general information (#65)

Co-authored-by: Raffique <raffiquemuir@yahoo.com>
Co-authored-by: Andre Vidal <53495324+andre-vidal@users.noreply.github.com>

andre-vidal committed:
Update deploy-documentation.yml

andre-vidal committed:
Update deploy-documentation.yml

Raffique-RD committed:
Sepa 735 documention for entities (#66)

* update entities page

* changed title

---------

Co-authored-by: Raffique <raffiquemuir@yahoo.com>

Raffique-RD committed:
update dotnet endpoints page (#67)

Co-authored-by: Raffique <raffiquemuir@yahoo.com>

Raffique-RD committed:
update dotnet query page (#68)

Co-authored-by: Raffique <raffiquemuir@yahoo.com>

Raffique-RD committed:
Sepa 737 documentation for commands (#69)

* updated commands page

* updates

---------

Co-authored-by: Raffique <raffiquemuir@yahoo.com>

tbennett109 committed:
Sepa 770 add conditional logic to fail gracefully if the dotnet commands dont execute as expected (#63)

* try catch added and conditional statement adjusted

* throw error if code ! = 0

* typo corrected

* use try catch and throw error for success1 variable

---------

Co-authored-by: tbennett109 <tbennett@ealdecoy.com>

Raffique-RD committed:
deleted unncessary pages (#70)

Co-authored-by: Raffique <raffiquemuir@yahoo.com>

Raffique-RD committed:
Sepa 738 documention for queries (#72)

* update dotnet query page

* update queries page

* update page documentation

* update documentation

---------

Co-authored-by: Raffique <raffiquemuir@yahoo.com>
Co-authored-by: Raffique Muir <git@VEN-MuirR.com>

Raffique-RD committed:
Sepa 737 documentation for commands (#71)

* updated commands page

* updates

* update commands page

* update page documentation

* fix: update typo

---------

Co-authored-by: Raffique <raffiquemuir@yahoo.com>
Co-authored-by: Raffique Muir <git@VEN-MuirR.com>
Co-authored-by: andre-vidal <avidal@realdecoy.com>

Raffique-RD committed:
Sep 736 documentation for endpoints (#73)

* update dotnet endpoints page

* update endpoints webpage

* update documentation

---------

Co-authored-by: Raffique <raffiquemuir@yahoo.com>
Co-authored-by: Raffique Muir <git@VEN-MuirR.com>

Raffique-RD committed:
Sepa 735 documention for entities (#74)

* update entities page

* changed title

* update entity website page

* update documentation

* update

---------

Co-authored-by: Raffique <raffiquemuir@yahoo.com>
Co-authored-by: Raffique Muir <git@VEN-MuirR.com>

andre-vidal committed:
hotfix: update features documentation (#75)

Davane committed:
Update deploy-package.yml

Davane committed:
Updated deploy-package.yml and fixed the manual trigger input label

ttaylor92 committed:
implemented scaffolding updates (#78)

Davane committed:
changed the way in which we deploy to NPM by updating the deploy-pack… (#79)

* changed the way in which we deploy to NPM by updating the deploy-package.yml

* Update deploy-package.yml

* updated the install dependencies command

* exposed version number and message update in github actions

* Fixed bug with versioning the npm package. Prevent the creation of git commits

* Added build step to the deploy-package.yml

Davane committed:
Dockerize commands for the dotnet namespace (#77)

* Updated the create-project command so that is uses docker instead of the host system tools when creating a project

* updated the "up" command to use docker instead of host dotnet CLI

* clean up the create project command

* So added the migration commands to be dependent on docker only. Added environment, configuration and continaer flags

* corrected the new step instructions after project

* added platform specific paths to command execution for project creation

* FIx "sh" executable on widows

* Removed unnecessary comments and logs

* dockerize down command

* Updated the documentation for the the dotnet namespace

* Fixed invalid filename when parsing migrations in the dotnet down command

---------

Co-authored-by: Davane Davis <ddavis@realdecoy.com>

Davane committed:
Updated the CI Pipeline to execute successfully using the UI inputs (#80)

* changed the way in which we deploy to NPM by updating the deploy-package.yml

* Update deploy-package.yml

* updated the install dependencies command

* exposed version number and message update in github actions

* Fixed bug with versioning the npm package. Prevent the creation of git commits

* Added build step to the deploy-package.yml

* Clean up

* Update deploy-package.yml

---------

Co-authored-by: Davane Davis <ddavis@realdecoy.com>

Davane committed:
Removed the branch triggers for the Github Actions (#81)

* changed the way in which we deploy to NPM by updating the deploy-package.yml

* Update deploy-package.yml

* updated the install dependencies command

* exposed version number and message update in github actions

* Fixed bug with versioning the npm package. Prevent the creation of git commits

* Added build step to the deploy-package.yml

* Clean up

* Update deploy-package.yml

* Removed the branch triggers for the Github actions pipeline

* Removed the isBeta check from the publish check in Actions

---------

Co-authored-by: Davane Davis <ddavis@realdecoy.com>

ttaylor92 committed:
implemented functionality (#82)

ttaylor92 committed:
Sdk 49 file changes for mobile (#84)

* Update path generation needed for scaffolding features

* updated constant naming

* removed unused logs

ttaylor92 committed:
Updated documentation to match SDK 50 (#87)

natvrey committed:
remove spectre references, start fixing typos (#92)

marlonajgayle committed:
Update document for dotnent overview and next steps (#93)

marlonajgayle committed:
Update project name instructions to use camel case as the project name format (#94)

Co-authored-by: Andre Vidal <53495324+andre-vidal@users.noreply.github.com>

andre-vidal committed:
Update deploy-documentation.yml

andre-vidal committed:
Sepa 785 implement vue3 feature support (#91)

* Updated to include support for vue3 plugins and localization updates

* Vue Changes:
- Updated the injection of localization plugin into the main.ts file
- Removed unnecessary code for installing Buefy and Vuetify plugins
- Updated the main.ts file to use the new injection methods for Vue 2 and Vue 3
- Added the injection of routes to the main router
- Refactored imports for vue2 methods and implementation and vue 3

* Update package.json and tsconfig.tsbuildinfo

- Update "bin" property in package.json to point to "./bin/run"
- Remove unnecessary "No newline at end of file" in package.json
- No significant changes in tsconfig.tsbuildinfo

* Update installation and serve commands

- Update the installation command to remove the flag '--legacy-peer-deps' in order to install dependencies correctly.
- Update the serve command to run 'npm run dev' instead of 'npm run serve' for serving the project.

This commit updates the installation and serve commands in the '3.next-steps.md' file under 'Vue/2.getting-started' directory. The installation command no longer includes the '--legacy-peer-deps' flag, ensuring correct installation of dependencies. The serve command has been updated to use 'npm run dev' instead of 'npm run serve' for serving the project. These changes improve the clarity and accuracy of the instructions provided in the documentation.

* Add route to router configuration when adding a new page

This commit adds a route object to the router configuration when a new page is added using the Vue CLI. The route object includes the path, name, component, and webpackChunkName properties. After creating the route object, it is injected into the router configuration using the injectRoutesIntoRouter function. This ensures that the newly added page is accessible through the specified path.

* Update package version and fix tsconfig.tsbuildinfo.

- Updated package version in package.json from "0.0.2-beta.12" to "0.0.2-beta.13".
- Fixed issue in tsconfig.tsbuildinfo by correcting the file paths.

For more details, see the specific changes made in each file.

* Removed console log and updated condition for checking presets

* SEPA-849: initial implementation of sentry integration (#76)

* initial implementation of sentry integration

* Added Sentry functionality as a preset during the project creation

* Rearrange Sentry prompt and loggging

* Added prompt for dsn to sentry plugin when ran standalone

---------

Co-authored-by: Davane Davis <ddavis@realdecoy.com>

* Added ability to import prebuilt ui components from .rdvue (#85)

* Added ability to import prebuilt ui components from .rdvue

* Refactor component command to update package.json and install dependencies in parallel

* Removed usage of exit()

* update copy files method and update packages

* docs: Update the docs to reflect migration from Vuex to Pinia for state management.

* docs:  Refactor and expand Vue service documentation

* docs: Update Vue pages doc with CLI and router instructions

* docs: Add comprehensive layout documentation & CLI usage

---------

Co-authored-by: Tyrone Taylor <28680107+ttaylor92@users.noreply.github.com>
Co-authored-by: Tahjyei Thompson <tahjyeithompson@gmail.com>
Co-authored-by: Mickey <31347338+JamaicanFriedChicken@users.noreply.github.com>
Co-authored-by: Davane Davis <ddavis@realdecoy.com>
Co-authored-by: nxtedecoy <107061280+nxtedecoy@users.noreply.github.com>

mitziemill committed:
Adding a page that will break down the folder structure that is generated in the .net template. (#95)

* Adding an overview/explanation of the folder structure used in dotnet template

* Update formatting of page to be less jumbled

* Update 7.folder structure.md

* Update formatting 7.folder structure.md

* Update 7.folder structure.md

Tajcore committed:
docs:Update documentation on vue components (#96)

Tajcore committed:
docs:Update documentation to remove references to RDVUE where necessary (#97)

Co-authored-by: Andre Vidal <53495324+andre-vidal@users.noreply.github.com>

Tajcore committed:
docs:Update documentation to include layout on CLI commands usage list (#98)

andre-vidal committed:
Sepa 785 implement vue3 feature support (#99)

* Updated to include support for vue3 plugins and localization updates

* Vue Changes:
- Updated the injection of localization plugin into the main.ts file
- Removed unnecessary code for installing Buefy and Vuetify plugins
- Updated the main.ts file to use the new injection methods for Vue 2 and Vue 3
- Added the injection of routes to the main router
- Refactored imports for vue2 methods and implementation and vue 3

* Update package.json and tsconfig.tsbuildinfo

- Update "bin" property in package.json to point to "./bin/run"
- Remove unnecessary "No newline at end of file" in package.json
- No significant changes in tsconfig.tsbuildinfo

* Update installation and serve commands

- Update the installation command to remove the flag '--legacy-peer-deps' in order to install dependencies correctly.
- Update the serve command to run 'npm run dev' instead of 'npm run serve' for serving the project.

This commit updates the installation and serve commands in the '3.next-steps.md' file under 'Vue/2.getting-started' directory. The installation command no longer includes the '--legacy-peer-deps' flag, ensuring correct installation of dependencies. The serve command has been updated to use 'npm run dev' instead of 'npm run serve' for serving the project. These changes improve the clarity and accuracy of the instructions provided in the documentation.

* Add route to router configuration when adding a new page

This commit adds a route object to the router configuration when a new page is added using the Vue CLI. The route object includes the path, name, component, and webpackChunkName properties. After creating the route object, it is injected into the router configuration using the injectRoutesIntoRouter function. This ensures that the newly added page is accessible through the specified path.

* Update package version and fix tsconfig.tsbuildinfo.

- Updated package version in package.json from "0.0.2-beta.12" to "0.0.2-beta.13".
- Fixed issue in tsconfig.tsbuildinfo by correcting the file paths.

For more details, see the specific changes made in each file.

* Removed console log and updated condition for checking presets

* SEPA-849: initial implementation of sentry integration (#76)

* initial implementation of sentry integration

* Added Sentry functionality as a preset during the project creation

* Rearrange Sentry prompt and loggging

* Added prompt for dsn to sentry plugin when ran standalone

---------

Co-authored-by: Davane Davis <ddavis@realdecoy.com>

* Added ability to import prebuilt ui components from .rdvue (#85)

* Added ability to import prebuilt ui components from .rdvue

* Refactor component command to update package.json and install dependencies in parallel

* Removed usage of exit()

* update copy files method and update packages

* docs: Update the docs to reflect migration from Vuex to Pinia for state management.

* docs:  Refactor and expand Vue service documentation

* docs: Update Vue pages doc with CLI and router instructions

* docs: Add comprehensive layout documentation & CLI usage

* update dotnet template tag

---------

Co-authored-by: Tyrone Taylor <28680107+ttaylor92@users.noreply.github.com>
Co-authored-by: Tahjyei Thompson <tahjyeithompson@gmail.com>
Co-authored-by: Mickey <31347338+JamaicanFriedChicken@users.noreply.github.com>
Co-authored-by: Davane Davis <ddavis@realdecoy.com>
Co-authored-by: nxtedecoy <107061280+nxtedecoy@users.noreply.github.com>

forevergareth committed:
added license and release v1 (#100)

Co-authored-by: Gareth Thomas <gthomas@realdecoy.com>

marlonajgayle committed:
update dotnet template shortname (#101)
